README.md
# got

A tool to make it easier to manage multiple code repositories using different VCSen

# VERSION [![CPAN version](https://badge.fury.io/pl/App-GitGot.svg)](http://badge.fury.io/pl/App-GitGot)

# BUILD INFO

[![Build Status](https://travis-ci.org/genehack/app-gitgot.svg?branch=main)](https://travis-ci.org/genehack/app-gitgot)
[![Coverage Status](https://coveralls.io/repos/genehack/app-gitgot/badge.svg?branch=main)](https://coveralls.io/r/genehack/app-gitgot?branch=main)

# USAGE

See usage information [on CPAN](https://metacpan.org/pod/distribution/App-GitGot/bin/got)


Changes
Changes file for {{$dist->name}}

{{$NEXT}}

1.339     2020-10-29 20:36:17-07:00 America/Los_Angeles

  * s/main/master/g in tests and documentation [#74] (genehack)

1.338     2020-10-26 18:46:10-07:00 America/Los_Angeles

  * switch JSON backend to JSON::MaybeXS (Karen Etheridge)
  * s/master/main/g in tests and documentation [#74] (genehack)

1.337     2020-04-25 15:33:17-07:00 America/Los_Angeles

  * Add 'readd' command to update repo entries in ~/.gitgot based on
    what is in working trees (genehack)
  * Add .editorconfig (genehack)
  * Add `--recursive` option to `clone` command (genehack,
    reimplementing yanick's code from #69) [#68]
  * switch from File::HomeDir to File::HomeDir::Tiny, for fewer
    dependencies (Karen Etheridge) [#70]

1.336     2018-04-13 15:38:00-07:00 America/Los_Angeles

  * Add 'git-got' binary to allow running as `git got` (genehack)

1.335     2018-03-19 21:10:19-07:00 America/Los_Angeles

  * Don't let the user pick a directory that already exists for
    got-clone; [#66] (Yanick)

1.334     2017-11-05 11:55:34-08:00 America/Los_Angeles
  * switch from using List::AllUtils to List::Util (Karen Etheridge)

  * Add `got milk` subcommand (Lucy Wyman)

  * Tweak detection of clean repos since git 2.15.0 changed the output.

1.333     2016-04-22 09:47:11-07:00 America/Los_Angeles

  * Add `--show-branch` option to `update-status` too

1.332     2016-04-17 13:53:37-04:00 America/New_York

  * Fix Path::Tiny dep in got-complete. Thanks @perlpunk [#58]
  * Add `--show-branch` option for `got status` [@perlpunk] [#59]

1.331     2016-01-03 20:30:14-08:00 America/Los_Angeles (TRIAL RELEASE)

  * Depend on Path::Tiny 0.072. Thx to SawyerX for pointing out this issue.

1.330     2015-04-14 21:27:19-07:00 America/Los_Angeles

  * No changes from last trial release.
  * No new features since last stable release.

1.32      2015-04-10 11:23:50-07:00 America/Los_Angeles (TRIAL RELEASE)

  * Update a number of tests to pass under Windows
  * Replace ENV{HOME} with File::HomeDir everywhere
  * Depend on a version of Git::Wrapper that doesn't use File::pushd

1.31      2015-04-05 21:42:56-07:00 America/Los_Angeles (TRIAL RELEASE)

  * Tweak some test code to maybe not blow up on Windows (John
    Anderson)
  * Set a minimum version of MooX::HandlesVia to avoid test failures
    (John Anderson)

1.30      2015-03-27 21:25:15-07:00 America/Los_Angeles

  * Make 'got fork' add the forked repo as remote 'upstream' unless
    -N/--noremoteadd option is given. (John Anderson)
  * See also changes for dev releases v1.25 through v1.29

1.29      2015-03-26 11:58:24-07:00 America/Los_Angeles (TRIAL RELEASE)

  * Docs for 'clone' and 'fork'. (John Anderson)
  * Be a bit more lenient in parsing GitHub URLs (John Anderson)
  * Make 'fork' more verbose [#51] (John Anderson)
  * Tweak 'clone' to be a bit better (John Anderson)

1.28      2015-03-23 22:32:27-07:00 America/Los_Angeles (TRIAL RELEASE)

  * Tweak missing directory error detection regex (John Anderson)

1.27      2015-03-22 19:46:53-07:00 America/Los_Angeles (TRIAL RELEASE)

  * Add some safety code to prevent mux from opening errything (John Anderson)
  * Add '-e' flag to 'tmux' subcommand [#41] (John Anderson)

1.26      2015-03-18 22:04:39-07:00 America/Los_Angeles (TRIAL RELEASE)

  * Lock Moo to version 2 or greater (John Anderson)
  * Report Git version being used in test suite (John Anderson)

1.25      2015-03-17 20:51:55-07:00 America/Los_Angeles

  * Fix failing tests. (Thanks Travis CI!) (John Anderson)

1.24      2015-03-16 16:56:55-07:00 America/Los_Angeles

  * Update copyright to 2015 (John Anderson)
  * Lock Net::GitHub to at least v0.74 [#47] (John Anderson)
  * Various testing improvements (John Anderson)
  * Turn off IO::Page in 'got all' (John Anderson)
  * Get rid of Term::Readline in favor of IO::Prompt::Simple (John Anderson)
  * Move towards using Path::Tiny for errything (John Anderson)
  * Whitespace cleanup (John Anderson)
  * Clean up 'use' stanzas (John Anderson)
  * delete 'version' command [#48] (John Anderson)

1.23      2015-03-15 01:50:42-07:00 America/Los_Angeles (TRIAL RELEASE)

  * Port to Moo (John Anderson)
  * Silence warning if PERL5LIB is not defined (Yanick Champoux)
  * Specify minimum version on List::AllUtils (John Anderson)

1.22      2015-02-26 07:40:05-08:00 America/Los_Angeles

  * Really fixed test failures (John Anderson)

1.21      2015-02-24 19:15:58-08:00 America/Los_Angeles

  * Fixed test failures (John Anderson)

1.20      2015-02-23 17:51:14-08:00 America/Los_Angeles

  * Add 'lib' subcommand (Yanick Champoux)
  * Add 'tag' subcommand (Yanick Champoux)
  * Add 'do' subcommand (Yanick Champoux)
  * Add '--json' option to 'got list' (Yanick Champoux)
  * Extend the 'add' subcommand in a number of ways:
      * Allow to pass more than one directory as arguments.
      * Add '--recursive', to hunt down git repos under a main
        directory.
      * Make '--tag X' set the default tag(s) to X
      * If there's only one remote, don't ask, use that one.
      * Instead of asking the user to type the remote URI, give the
        choice between the remote name/URL pairs.

1.19      2014-08-29 22:07:03-07:00 America/Los_Angeles

  * Make 'mux' work with multiple repos instead of just one
  * Add 'dirty' flag to 'mux' to make it open all repos with unclear
    status

1.18      2014-08-12 22:03:34-05:00 America/Chicago

  * Documentation updates for 'fetch' and 'gc' (Dan McCracken)

1.17      2014-07-24 10:03:54-07:00 America/Los_Angeles

  * Modify dist.ini to pull the INSTALL file from the build back into
    the repo (addresses #28)

1.16      2014-07-23 16:19:42-07:00 America/Los_Angeles

  * Load Net::GitHub on the fly in the 'fork' command. Make it an
    optional dep. Provide nice error message if it is not installed
    when the 'fork' command is used.

1.15      2014-05-08 21:19:33-07:00 America/Los_Angeles


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.015.
use strict;
use warnings;

use 5.014;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "A tool to make it easier to manage multiple git repositories.",
  "AUTHOR" => "John SJ Anderson <john\@genehack.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "App-GitGot",
  "EXE_FILES" => [
    "bin/git-got",
    "bin/got",
    "bin/got-complete"
  ],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.014",
  "NAME" => "App::GitGot",
  "PREREQ_PM" => {
    "App::Cmd::Setup" => 0,
    "Capture::Tiny" => 0,
    "Class::Load" => 0,
    "Config::INI::Reader" => 0,
    "Cwd" => 0,
    "Data::Dumper" => 0,
    "File::Copy::Recursive" => 0,
    "File::HomeDir::Tiny" => 0,
    "File::chdir" => 0,
    "Git::Wrapper" => "0.042",
    "IO::Prompt::Simple" => 0,
    "List::Util" => "1.45",
    "Moo" => "2.000000",
    "MooX::HandlesVia" => "0.001008",
    "Path::Tiny" => "0.072",
    "PerlX::Maybe" => 0,
    "Term::ANSIColor" => 0,
    "Test::MockObject" => 0,
    "Try::Tiny" => 0,
    "Type::Library" => 0,
    "Type::Utils" => 0,
    "Types::Standard" => 0,
    "YAML" => 0,
    "autodie" => 0,
    "namespace::autoclean" => 0,
    "overload" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "App::Cmd::Tester" => 0,
    "Carp" => 0,
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::Class" => 0,
    "Test::Class::Load" => 0,
    "Test::Exception" => 0,
    "Test::File" => 0,
    "Test::More" => "0.94",
    "lib" => 0,
    "parent" => 0,
    "strict" => 0
  },
  "VERSION" => "1.339",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "App::Cmd::Setup" => 0,
  "App::Cmd::Tester" => 0,
  "Capture::Tiny" => 0,
  "Carp" => 0,
  "Class::Load" => 0,
  "Config::INI::Reader" => 0,
  "Cwd" => 0,
  "Data::Dumper" => 0,
  "File::Copy::Recursive" => 0,
  "File::HomeDir::Tiny" => 0,
  "File::Spec" => 0,
  "File::Temp" => 0,
  "File::chdir" => 0,
  "Git::Wrapper" => "0.042",
  "IO::Handle" => 0,
  "IO::Prompt::Simple" => 0,
  "IPC::Open3" => 0,
  "List::Util" => "1.45",
  "Moo" => "2.000000",
  "MooX::HandlesVia" => "0.001008",
  "Path::Tiny" => "0.072",
  "PerlX::Maybe" => 0,
  "Term::ANSIColor" => 0,
  "Test::Class" => 0,
  "Test::Class::Load" => 0,
  "Test::Exception" => 0,
  "Test::File" => 0,
  "Test::MockObject" => 0,
  "Test::More" => "0.94",
  "Try::Tiny" => 0,
  "Type::Library" => 0,
  "Type::Utils" => 0,
  "Types::Standard" => 0,
  "YAML" => 0,
  "autodie" => 0,
  "lib" => 0,
  "namespace::autoclean" => 0,
  "overload" => 0,
  "parent" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
